integrationTests/chainSimulator/mempool/benchmarks_test.go

```diff
@@ -27,59 +31,156 @@ var (
 	}
 )
 
+func setUpPprofFilesIfNecessary(t *testing.T) (bool, func(t *testing.T) *os.File, func(t *testing.T, f *os.File)) {
+	return shouldCreatePprofFiles(), createPprofFiles, shouldStopProfiling
```

Comment:
I would handle here the logic and only return 2 methods

suggested changes:
- rename `shouldStopProfiling` to `stopProfiling`
- `setUpPprofFilesIfNecessary` only returns 2 functions, `func(t *testing.T) io.Closer`, `func(t *testing.T, closable io.Closer))`
- if `shouldCreatePprofFiles` returns false, then simply return empty methods that do nothing/call mocked/disabled components
- `stopProfiling` receives a Closer interface parameter, instead of pointer to file
- `createPprofFiles` returns a Closer interface, instead of file pointer. If env var is not found, then return a CloserMock/CloserStub that do nothing

This way we reduce the check for `shouldDoProfiling`